services/user/rename.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package user
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/services/agit"
+	container_service "code.gitea.io/gitea/services/packages/container"
+)
+
+func renameUser(ctx context.Context, u *user_model.User, newUserName string) (err error) {
+	if u.IsOrganization() {
+		return fmt.Errorf("cannot rename organization")
+	}
+
+	if u.LowerName == strings.ToLower(newUserName) {
+		return fmt.Errorf("new username is the same as the old one")
+	}
+
+	if err := user_model.ChangeUserName(u, newUserName); err != nil {
```

Comment:
Then we can have a `db.WithTx` to begin a new database transaction.

Reply:
moved into a transaction

Reply:
need ctx here

Reply:
added :)

Reply:
I mean we need to change the function to user_model.ChangUseName the first parameter to Ctx?

Reply:
Yes, please refresh page and you can see that it is as you requested:)

Reply:
Sorry, I have refreshed many times. It's the same. Shouldn't this be

```suggestion
	if err := user_model.ChangeUserName(ctx, u, newUserName); err != nil {
```